--- /NFS/home/AFoglia/myroot/stow/doxypy-0.3rc1-ajf2/bin/doxypy.py    2007-12-18 11:43:58.000000000 -0500
+++ /NFS/home/AFoglia/myroot/stow/doxypy-0.3rc1-ajf3/bin/doxypy.py    2007-12-18 12:08:12.000000000 -0500
@@ -201,6 +201,15 @@
        self.comment = []
        self.output.append(self.fsm.current_input)

+    def _docstringSummaryToBrief(self, line):
+        """     Adds \brief to the a docstring summary line.
+                A \brief is prepended, provided no other doxygen command is at the start of the line.
+        """
+        if not (line.strip())[0] in ('@', '\\') :
+            return "\\brief "+line
+        else :
+            return line
+       
    def appendCommentLine(self, match):
        """    Appends a comment line.
            The comment delimiter is removed from multiline start and ends as
@@ -214,7 +223,10 @@
            # remove comment delimiter from begin and end of the line
            activeCommentDelim = match.group(1)
            line = self.fsm.current_input
-            self.comment.append(line[line.find(activeCommentDelim)+len(activeCommentDelim):line.rfind(activeCommentDelim)])
+            comment_contents = line[line.find(activeCommentDelim)+len(activeCommentDelim):line.rfind(activeCommentDelim)]
+            if (options.autobrief):
+                comment_contents = self._docstringSummaryToBrief(comment_contents)
+            self.comment.append(comment_contents)
            if (to_state == "DEFCLASS_BODY"):
                self.__closeComment()
                self.defclass = []
@@ -237,6 +249,9 @@
        else:
            # just append the comment line
            self.comment.append(self.fsm.current_input)
+        if (options.autobrief):
+            if (len(self.comment) == 2 and self.comment[1].strip() == ''):
+                    self.comment[0] = self._docstringSummaryToBrief(self.comment[0])
    
    def appendNormalLine(self, match):
        """ Appends a line to the output. """
@@ -309,6 +324,11 @@
        action="store_true", dest="strip",
        help="enables trimming of docstrings, might be useful if you get oddly spaced output"
    )
+    parser.add_option("--autobrief",
+        action="store_true", dest="autobrief",
+        help="Use docstring summary line as brief descriptions"
+    )
+                  
    ## parse options
    global options 
